In [1]:
import pandas as pd

In [2]:
from PIL import Image
import matplotlib.pyplot as plt
import numpy as np

In [10]:
image_path = '../../data/HAM10000_images_part_1/ISIC_0024315.jpg'  # Replace with the actual path to your JPG file

try:
    # Load the image using Pillow
    img = Image.open(image_path)

    # Convert the image to a NumPy array for Matplotlib
    img_array = np.array(img)


    print(img_array.shape)
    # Plot the image using Matplotlib
    # plt.imshow(img_array)
    # # plt.axis('off')  # Hide the axes for a cleaner image display
    # plt.title('Loaded JPG Image')
    # plt.show()

except FileNotFoundError:
    print(f"Error: The file '{image_path}' was not found.")
except Exception as e:
    print(f"An error occurred: {e}")

(450, 600, 3)


Images apear are (450 x 600) with RBG channels. These arrays are the inputs into the CNN's. The difficult part of what we are going to need to do for the preprocessing is creating a matching label for each loaded image. We will need to define a nice class that process all of this. Further, we need to implement the seperation of training and testing data. 

In reality, this we will be pretty simple to figure out a way to make it work.

For a later on ease we should usr pytorches dataloader to make the implementation a lot easier. 

#### Labels are found in the metadeta csv file. 
We are going to use pandas to load in the CSV file and work through all the labels. 

In the HAMM10000 data set there are a lot of different labels. "MEL NV BCC AKIEC BKL DF VASC" are all of them. The SkinCancer dataset only has malignant and benign as labels. I think that for our labels we give the following, safe or unsafe. This will just straightenup what we are doing. 